### PR TITLE
chore: improve logger middleware to better visibility

### PR DESCRIPTION
**Pull request by ezr-ondrej**

### CODE REVIEW

File: internal/middleware/logger.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ func LoggerMiddleware(rootLogger *zerolog.Logger) func(next http.Handler) http.H
 				Logger()
 			t1 := time.Now()
 
+			logger.Debug().Msg("Started request")
```
Comment (ezr-ondrej) [inline internal/middleware/logger.go:34]: I think method and URL is in the annotation vars already, so nothing on top of that is needed I guess.

Comment (lzap) [inline internal/middleware/logger.go:34]:
```suggestion
			logger.Debug().Msgf("Started %s request %s", r.Method, r.URL.Path)
```

File: internal/middleware/logger.go [hunk 2]
```diff
@@ -49,7 +51,7 @@ func LoggerMiddleware(rootLogger *zerolog.Logger) func(next http.Handler) http.H
 
 				log.Info().
 					Int("status", ww.Status()).
-					Msg(fmt.Sprintf("Completed %s request %s in %s ms with %d",
+					Msg(fmt.Sprintf("Completed %s request %s in %s with %d",
```
Comment (lzap) [inline internal/middleware/logger.go:54]:
```suggestion
					Msgf("Completed %s request %s in %s with %d",
```

File: internal/middleware/logger.go (hunk 1 above)
Comment (lzap) [reply internal/middleware/logger.go:34]: Yes it is and the general idea of structured logging is to only put message without copies of these fields, yet, I would suggest to format it the same way as in "Completed..." message, I find it very useful when searching logs by eyes without all the fields enabled in Kibana.

Comment (lzap) [reply internal/middleware/logger.go:34]: I find it very useful, this is subjective I have to admit:

image

Comment (ezr-ondrej) [reply internal/middleware/logger.go:34]: What for? we have these in the logger itself :thinking:

Comment (ezr-ondrej) [reply internal/middleware/logger.go:34]: see my comment ^^ :)

Comment (lzap) [reply internal/middleware/logger.go:34]: I pasted an example screenshot above, see how useful this is. I would argue that we already have this for "Completed" message, for consistency let's please keep it the same way. I find it very useful myself.

In general, I agree this should not be the way we create logs, but this middleware is special - all requests have these logs.

File: internal/middleware/logger.go [hunk 3]
```diff
@@ -31,6 +30,9 @@ func LoggerMiddleware(rootLogger *zerolog.Logger) func(next http.Handler) http.H
 				Logger()
 			t1 := time.Now()
 
+			lHeaders := logger.With().Fields(r.Header).Logger()
+			lHeaders.Debug().Msg("Started request with above headers")
```
Comment (lzap) [inline internal/middleware/logger.go:34]: There is no need to copy logger just to use it once, why don't you just do:

```
logger.Debug().Fields(r.Header).Msgf("Started %s request %s", r.Method, r.URL.Path")
```

I am asking once again for the same formatting as we do for the "Completed" message so I can "pair" the two by eyes in Kibana.

Comment (lzap) [reply internal/middleware/logger.go:34]: I do not know how to create a submap, I think this is okay.

File: internal/middleware/logger.go (hunk 1 above)
Comment (ezr-ondrej) [reply internal/middleware/logger.go]: oki, you're right it doesn't hurt :)

File: internal/middleware/logger.go (hunk 3 above)
Comment (ezr-ondrej) [reply internal/middleware/logger.go]: oh I missed this comment, there is usage of method on a pointer and for some reason it doesn't work in-line (no idea what went wrong there, didn't investigate).

**Review by lzap — APPROVED** [review]
Thanks, I appreciate having method and path in both messages, this allows me to quickly find proper requests without bringing additional columns in Kibana.

**Pull request merged**